techs/files.js

```diff
@@ -106,39 +106,42 @@ module.exports = inherit(require('enb/lib/tech/base-tech.js'), {
             .spread(function (data, levels) {
                 return requireSourceDeps(data, depsFilename)
                     .then(function (sourceDeps) {
-                        var fileList = new FileList(),
-                            dirList = new FileList(),
-                            files = {},
-                            dirs = {};
+                        var fileHash = {},
+                            dirHash = {},
+                            fileList = new FileList(),
+                            dirList = new FileList();
 
                         for (var i = 0, l = sourceDeps.length; i < l; i++) {
                             var dep = sourceDeps[i],
                                 entities;
+
                             if (dep.elem) {
                                 entities = levels.getElemEntities(dep.block, dep.elem, dep.mod, dep.val);
                             } else {
                                 entities = levels.getBlockEntities(dep.block, dep.mod, dep.val);
                             }
 
-                            addEntityFiles(entities);
+                            addEntities(entities);
                         }
 
-                        fileList.addFiles(Object.keys(files).map(function (filename) {
-                            return files[filename];
-                        }));
+                        function addEntities(entities) {
+                            fileList.addFiles(entities.files.filter(function (file) {
```

Comment:
isn't it possible to store files and dirs in the same hash and as so reuse filter callback for both cases?